examples/with-sentry/pages/_app.js

```diff
@@ -1,23 +1,32 @@
 import App from 'next/app'
-import * as Sentry from '@sentry/browser'
+import { captureException } from '../utils/sentry'
 
-const SENTRY_PUBLIC_DSN = ''
+class MyApp extends App {
+  // This reports errors before rendering, when fetching initial props
+  static async getInitialProps (appContext) {
+    const { Component, ctx } = appContext
 
-export default class MyApp extends App {
-  constructor (...args) {
-    super(...args)
-    Sentry.init({dsn: SENTRY_PUBLIC_DSN})
+    let pageProps = {}
+
+    try {
+      if (Component.getInitialProps) {
+        pageProps = await Component.getInitialProps(ctx)
+      }
+    } catch (e) {
+      captureException(e, ctx)
```

Comment:
This should be moved to _error.js instead, we use something like this for zeit.co:

```
  static getInitialProps({ res, err }) {
    console.log({err})
    const statusCode = res ? res.statusCode : err ? err.statusCode : null
    Sentry.captureException(err)
    return { statusCode }
  }
```

Reply:
It doesn't really work and results in Synthetic errors. ctx is passed to 1st party captureException from which relevant data is extracted. I don't see how it's security issue

Reply:
I tried lots of different ways to integrate sentry in the _error.js but all of them resulted in Synthetic errors